plugin/dyf/Spatial Daylight Autonomy.dyf

```diff
@@ -16,23 +16,18 @@ per year. The sDA value is expressed as a percentage of area." ID="695e3d7e-fe8e
 &quot;&quot;&quot;Find packages from Dynamo folder and prepare the code.&quot;&quot;&quot;
 import clr
 from System import AppDomain, Environment, IO
+from System.Reflection import Assembly
 
 def tryGetPluginPath(plugin):
     &quot;&quot;&quot;Try to get path to plugin folder.&quot;&quot;&quot;
-    clr.AddReference('ProtoGeometry')
-    _loc = tuple(a.Location
-                 for a in AppDomain.CurrentDomain.GetAssemblies()
-                 if 'ProtoGeometry' in a.FullName)
-
-    try:
-        _ver = float(_loc[0].split('\\')[-2])
-    except:
-        _ver = float(_loc[0].split('\\')[-2].split(' ')[-1])
+    dynamo = Assembly.Load('DynamoCore')
+    dynamoV = dynamo.GetName().Version.ToString().split('.')
+    _ver = float(dynamoV[0] + '.' + dynamoV[1])
 
     assert _ver &gt;= 1.2, 'You need Dynamo 1.2 or higher to use this plugin!'
 
     _appdata = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
-    _dynamoPath = '%s\\%.1f' % (('\\').join(_loc[0].split('\\')[-4:-2]), float(_loc[0].split('\\')[-2]))
+    _dynamoPath = '%s\\%s\\%.1f' % ('Dynamo', 'Dynamo Core', _ver)
```

Comment:
Will this also work for Dynamo Revit? It looks like that it will only work for Dynamo Core.

Reply:
Uh, that's a good question, I thought I tested it in Dynamo Revit, but it's been a while, so maybe not....and needs to be adjusted....unfortunately, I've left Autodesk about a month ago to do a startup, and don't have access to Revit anymore to try it out :(

Reply:
Got it! Congrats for the new adventure.

I talked to the Dynamo team last week and hopefully they will make access to the document available from inside the Python node so we don't have to go through hacky solutions like this.

Also just in case and since I didn't know myself, you should apply for Autodesk small business partnership so you can get licenses for Autodesk product if needed.

Reply:
oh, I think in Dynamo Revit the path should be 'Dynamo Revit'.